spec/controllers/tag_wranglers_controller_spec.rb

```diff
@@ -17,4 +17,121 @@
       end
     end
   end
+
+  describe "#report_csv" do
+    shared_examples "prevents access to the report" do
+      it "redirects with an error" do
+        get :report_csv, params: { id: user.login }
+        it_redirects_to_with_error(root_url, "Sorry, only an authorized admin can access the page you were trying to reach.")
+      end
+    end
+
+    context "when not logged in" do
+      it_behaves_like "prevents access to the report"
+    end
+
+    context "when logged in as the wrangler" do
+      before { fake_login_known_user(user) }
+
+      it_behaves_like "prevents access to the report"
+    end
+
+    context "when logged in as an admin without proper authorization" do
+      before { fake_login_admin(admin) }
+
+      context "with no role" do
+        let(:admin) { create(:admin) }
+
+        it_behaves_like "prevents access to the report"
+      end
+
+      (Admin::VALID_ROLES - %w[superadmin tag_wrangling]).each do |admin_role|
+        context "with role #{admin_role}" do
+          let(:admin) { create(:admin, roles: [admin_role]) }
+
+          it_behaves_like "prevents access to the report"
+        end
+      end
+    end
+
+    context "when logged in as an admin with proper authorization" do
+      before { fake_login_admin(admin) }
+
+      %w[superadmin tag_wrangling].each do |admin_role|
+        context "with role #{admin_role}" do
+          let(:admin) { create(:admin, roles: [admin_role]) }
+
+          it "allows access to the report" do
+            get :report_csv, params: { id: user.login }
+            expect(response).to have_http_status(:success)
+          end
+
+          it "only includes wranling activity for the specified user" do
```

Comment:
Typo: should be "wrangling."

Reply:
Hah, I misspelled that word so many times, and I guess this one slipped through. Thanks!